### ISSUE TITLE
Link Map plugin does not work in Windows 10

### ISSUE THREAD

**Issue by frogmouse** [zim-desktop-wiki/zim-desktop-wiki#2012]
The Link Map plugin requires the external modules _graphviz_ and _xdot_ to be installed. Once this is done, Zim continues to report that _xdot_ doesn't exist.  This is true in the sense that _xdot_ is not functional -- it cannot find the module '_gi'_ from _graphviz_ -- though it reports no errors.

The [upstream developer is aware of this](https://github.com/jrfonseca/xdot.py/issues/70), has suggested a work-around that does not fix the problem for me, and has marked this problem "wontfix".

The plugin works as it should on linux, and is a valuable addition to Zim.   I'd like to be able to use it when running windows.

**Comment by fidthorn2vc** [comment]
If run the command "import xdot" in python , it says "ModuleNotFoundError: No module named 'numpy'".

Plugin worked for me after installation mingw-w64-x86_64-python-numpy package in msys2.

**Comment by introt** [comment]
@frogmouse does the plugin work after installing `mingw-w64-x86_64-python-numpy` as suggested by @fidthorn2vc ?

**Comment by frogmouse** [comment]
I'm not certain what this would do, but I haven't tried it -- I have no desire to install msys2, mingw, or cygwin on this box.  I did check out numpy and xdot in my Win10 python3 and got this result:

C:\Users\frogmouse>python
Python 3.8.2 (tags/v3.8.2:7b3ab59, Feb 25 2020, 23:03:10) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> import xdot
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\frogmouse\AppData\Roaming\Python\Python38\site-packages\xdot\__init__.py", line 26, in <module>
    from . import ui
  File "C:\Users\frogmouse\AppData\Roaming\Python\Python38\site-packages\xdot\ui\__init__.py", line 3, in <module>
    from .window import DotWidget, DotWindow
  File "C:\Users\frogmouseimport \AppData\Roaming\Python\Python38\site-packages\xdot\ui\window.py", line 25, in <module>
    import gi
ModuleNotFoundError: No module named 'gi'
>>>

My workaround is to have a directory shared between Windows and a VBox VM running the linux of my choice.  I can use do most of my editing in the Win Zim and get the link map to work in the linux instantiation.

Thanks for keeping an eye on this problem.

**Comment by introt** [comment]
My bad - we use MSYS2 for the Windows installer, so I imagined it sticking around. What I'm looking for is whether adding `mingw-w64-x86_64-python-numpy` to the Windows dependencies solves the problem.

>  This is true in the sense that xdot is not functional -- it cannot find the module 'gi' from graphviz -- though it reports no errors.

Not finding `gi` shouldn't be the issue here; Zim wouldn't start without its Gtk dependencies.

Could you try enabling the plugin and then post the logs from Tools > View debug log? The command prompt Python doesn't reflect Zim's environment.

**Comment by frogmouse** [comment]
OK.  First thing I got was an error box with the text, "No Application found for text/plain".  So I searched and found the place to define that application.  I find that there are 762 lines in the debug log -- including my efforts to define the text/plain application.  I'm going to try shutting zim down and restarting it, then loading the plugin -- hopefully that'll shrink the debug file.

It did.  Now it's 60 lines long.: 

[zim.log](https://github.com/zim-desktop-wiki/zim-desktop-wiki/files/9742163/zim.log)

I think the germane lines are:

DEBUG: Action: show_preferences
DEBUG: Loading description for plugin: arithmetic
DEBUG: Opening dialog "Preferences"
DEBUG: BackgroundCheck finished
DEBUG: Loading description for plugin: linkmap
DEBUG: Dialog response OK
DEBUG: Closed dialog "Preferences"

In between "loading description ..." and "Dialog response ... " I got this info box with the error Zim found highlighted in red:

![zim-debLog](https://user-images.githubusercontent.com/30908800/194770196-652e3724-267e-4563-a81f-194b6b5cc39f.JPG)

This is the first time I've tried to insert files into a comment.  I hope it worked satisfactorily.

Is there any other way I can help?

**Comment by frogmouse** [comment]
OOPS.  Wrong picture.  Here's the error box" 
![ZimLinkMapFail](https://user-images.githubusercontent.com/30908800/194770478-43c5866a-5d49-4620-9cd7-e92cf068eab0.JPG)

**Comment by menthalo37** [comment]
Hello
I meet same issue (xdot not found while tryinug to use either **textmindmap** or **linkmap** ZIM Notes plugins) under **Windows 10**. I want just to know if there is an existing way to to make it run (step by step please) or if this issue will need another Zim version to be fixed.

Thanks.
E.

**Comment by k4nzdroid** [comment]
OS: Windows 11
Zim: zim-desktop-wiki-0.75.1-setup-w64_x86

`xdot - Failed` means import xdot failed.

Add following code into plugin/linkmap.py to find out the ImportError:
```
try:
	import xdot
	from xdot import DotWidget
except ImportError as err:
	logger.debug("import error: %s", err)   // add
	xdot = None
	class DotWidget:  # workaround
		pass
```
rebuild, install, run and check the debug log, we got this error:
```
...
DEBUG: import error: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.10 from "C:\Program Files\Zim Desktop Wiki\zim.exe"
  * The NumPy version is: "1.24.3"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'

DEBUG: Loading description for plugin: arithmetic
...
```
I guess we have the same cause.

**Comment by k4nzdroid** [comment]
The zim-desktop-wiki-0.75.1-setup-w64_x86 is built with MSYS, and the mingw-w64-x86_64-python-numpy cause the problem.

To fix this issus, remove the mingw-w64-x86_64-python-numpy, install numpy from PyPi, and rebuild zim from source.

After that, The LinkMap will work.

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/40691275/f9cee0e2-47f7-4117-b46c-fe6f00b050e7)
